### ISSUE TITLE
Allow users to disable instrumentation of specific modules

### ISSUE THREAD

**Issue by watson** [elastic/apm-agent-nodejs#273]
We should allow the users to blacklist certain modules from being instrumented. This allows users to opt out in case instrumentation of certain modules cause problems or generate too much noisy data.

Suggestion for config option name: `disableInstrumentation`.

Usage example (inline config):

```js
require('elastic-apm-node').start({
  disableInstrumentation: ['mysql', 'ws']
})
```

Usage example (environment variables):

```
ELASTIC_APM_DISABLE_INSTRUMENTATION=mysql,ws
```

Notice the use of comma separated values in the environment variable. The config logic of the Node.js agent currently doesn't support this, but that should be simple to add.

**Comment by watson** [comment]
@elastic/apm-agent-devs let's align on this across all agents if possible

**Comment by beniwohli** [comment]
The Python agent already has this functionality, although it isn't really documented:

https://github.com/elastic/apm-agent-python/blob/99065fe525f7251003b2fb3eedc044e53f1d33df/elasticapm/instrumentation/packages/base.py#L50-L54

so e.g. setting a `SKIP_INSTRUMENT_MYSQL` environment variable would keep us from instrumenting MySQL connections.

**Comment by watson** [comment]
@beniwohli what's the reason for having a new config option per module instead of _one_ config option with a list of comma separated module names?

**Comment by beniwohli** [comment]
This is an option that's been in there from the very beginning, before we had a proper configuration system. That's also why it isn't prefixed with `ELASTIC_APM_` like all our other environment variables. I'm open to change it, as it currently isn't documented and hence not really covered by backwards compatibility guarantees IMO.

**Comment by mikker** [comment]
How about agreeing on `ELASTIC_APM_DISABLE_INTEGRATIONS=mysql,ws`? IMO integrations is a better word because it signals 3rd party more than _all instrumentation_. Also plural signals list. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Qard** [comment]
It's possible we might want module-specific configurations for other things in the future.

**Comment by mikker** [comment]
@Qard Those will probably be very language specific? Or at least too specific to warrant an ENV option?

**Comment by watson** [comment]
Hmm to me "integrations" doesn't sound right. We don't integrate with mysql, we instrument it. Making it plural sounds like a good idea though.

**Comment by mikker** [comment]
I might have the wrong understanding of what an _integration_ is 😀

**Comment by felixbarny** [comment]
I'd also prefer instrumentation rather than integration

**Comment by watson** [comment]
This have now been released in v1.7.0: https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#disable-instrumentations
